src/Types/PlutusAddress.purs

```diff
@@ -0,0 +1,192 @@
+module Types.PlutusAddress where
+
+import Prelude
+import Data.Maybe (Maybe(..))
+import Data.Array
+import Data.Tuple (Tuple(Tuple), fst)
+import Data.Generic.Rep (class Generic)
+import Data.Show.Generic (genericShow)
+import Data.Newtype (class Newtype, wrap)
+import Data.UInt (UInt, fromInt, toInt, (.&.), and, zshr, shl)
+import Types.ByteArray (ByteArray, byteArrayToIntArray, byteArrayFromIntArray)
+import Types.UnbalancedTransaction (PubKeyHash)
+import Types.Scripts (ValidatorHash)
+import Types.PlutusData (PlutusData(Constr, Map, List, Integer, Bytes))
+import Serialization.Address (Pointer, addressBytes)
+import Serialization.Address (Address) as Serialization
+import Serialization.Hash (ed25519KeyHashFromBytes, scriptHashFromBytes) as Hash
+import ToData (class ToData, toData)
+import FromData (class FromData, fromData)
+
+--------------------------------------------------------------------------------
+-- Credential
+--------------------------------------------------------------------------------
+
+data Credential
+  = PubKeyCredential PubKeyHash
+  | ScriptCredential ValidatorHash
+
+derive instance Generic Credential _
+instance Show Credential where
+  show = genericShow
+
+instance ToData Credential where
+  toData (PubKeyCredential pubKeyHash) =
+    Constr zero [ toData pubKeyHash ]
+  toData (ScriptCredential validatorHash) =
+    Constr one [ toData validatorHash ]
+
+instance FromData Credential where
+  fromData (Constr n [ pd ])
+    | n == zero = PubKeyCredential <$> fromData pd
+    | n == one = ScriptCredential <$> fromData pd
+  fromData _ = Nothing
+
+--------------------------------------------------------------------------------
+-- StakingCredential
+--------------------------------------------------------------------------------
+
+data StakingCredential
+  = StakingHash Credential
+  | StakingPtr Pointer
+
+derive instance Generic StakingCredential _
+
+instance Show StakingCredential where
+  show = genericShow
+
+instance ToData StakingCredential where
+  toData (StakingHash credential) =
+    Constr zero [ toData credential ]
+  toData (StakingPtr ptr) =
+    Constr one [ toData ptr.slot, toData ptr.txIx, toData ptr.certIx ]
+
+instance FromData StakingCredential where
+  fromData (Constr zero [ pd ]) =
+    StakingHash <$> fromData pd
+  fromData (Constr one [ slotD, txIxD, certIxD ]) =
+    StakingPtr <$> ({ slot: _, txIx: _, certIx: _} <$>
+      fromData slotD <*> fromData txIxD <*> fromData certIxD)
+  fromData _ = Nothing
+
+--------------------------------------------------------------------------------
+-- Address
+--------------------------------------------------------------------------------
+
+data PlutusAddress = Address
```

Comment:
I think the the type constructor should be `Address`, as this is the familiar representation for our users (compared to our other `Address`). I believe @vvtran was planning on re-organizing modules into `Cardano.*` and `Plutus.*` namespaces, so this could live in `Plutus.Types.Address` and be imported as `Plutus.Address` or something